sklearn/metrics/tests/test_score_objects.py

```diff
@@ -145,37 +149,105 @@ def test_all_scorers_repr():
         repr(scorer)
 
 
-def test_check_scoring():
-    # Test all branches of check_scoring
+def check_scoring_validator_for_single_metric_usecases(scoring_validator):
+    # Test all branches of single metric usecases
     estimator = EstimatorWithoutFit()
     pattern = (r"estimator should be an estimator implementing 'fit' method,"
                r" .* was passed")
-    assert_raises_regexp(TypeError, pattern, check_scoring, estimator)
+    assert_raises_regexp(TypeError, pattern, scoring_validator, estimator)
 
     estimator = EstimatorWithFitAndScore()
     estimator.fit([[1]], [1])
-    scorer = check_scoring(estimator)
+    scorer = scoring_validator(estimator)
     assert_true(scorer is _passthrough_scorer)
     assert_almost_equal(scorer(estimator, [[1]], [1]), 1.0)
 
     estimator = EstimatorWithFitAndPredict()
     estimator.fit([[1]], [1])
     pattern = (r"If no scoring is specified, the estimator passed should have"
                r" a 'score' method\. The estimator .* does not\.")
-    assert_raises_regexp(TypeError, pattern, check_scoring, estimator)
+    assert_raises_regexp(TypeError, pattern, scoring_validator, estimator)
 
-    scorer = check_scoring(estimator, "accuracy")
+    scorer = scoring_validator(estimator, "accuracy")
     assert_almost_equal(scorer(estimator, [[1]], [1]), 1.0)
 
     estimator = EstimatorWithFit()
-    scorer = check_scoring(estimator, "accuracy")
+    scorer = scoring_validator(estimator, "accuracy")
     assert_true(isinstance(scorer, _PredictScorer))
 
     estimator = EstimatorWithFit()
-    scorer = check_scoring(estimator, allow_none=True)
+    scorer = scoring_validator(estimator, allow_none=True)
     assert_true(scorer is None)
 
 
+def check_multimetric_scoring_single_metric_wrapper(*args, **kwargs):
```

Comment:
maybe a simple comment? what does this thing do? Remind me to review after you put some explanation here on what this is supposed to do ;)